src/Apps/Order/Components/AddressModal.tsx

```diff
@@ -108,185 +121,184 @@ export const AddressModal: React.FC<Props> = ({
 
-                if (userMessage) {
-                  actions.setFieldError(userMessage.field, userMessage.message)
-                } else {
-                  setCreateUpdateError(GENERIC_FAIL_MESSAGE)
-                }
-                actions?.setSubmitting(false)
-                onError && onError(message)
+              if (userMessage) {
+                actions.setFieldError(userMessage.field, userMessage.message)
+              } else {
+                setCreateUpdateError(GENERIC_FAIL_MESSAGE)
               }
+              actions?.setSubmitting(false)
+              onError && onError(message)
+            }
 
-              const handleSuccess = savedAddress => {
-                // update default address only if isDefault changed or new
-                // address marked ad default
-                if (
-                  values?.isDefault &&
-                  values?.isDefault !== address?.isDefault
-                ) {
-                  updateUserDefaultAddress(
-                    relayEnvironment,
-                    savedAddress?.createUserAddress?.userAddressOrErrors
-                      ?.internalID || address?.internalID,
-                    () => {
-                      onSuccess(savedAddress)
-                    },
-                    onError
-                  )
-                } else {
-                  onSuccess && onSuccess(savedAddress)
-                }
-
-                setCreateUpdateError(null)
+            const handleSuccess = savedAddress => {
+              // update default address only if isDefault changed or new
+              // address marked ad default
+              if (
+                savedAddress?.createUserAddress?.userAddressOrErrors
+                  ?.internalID &&
+                values?.isDefault &&
+                values?.isDefault !==
+                  (initialAddress as SavedAddressType)?.isDefault
+              ) {
+                updateUserDefaultAddress(
+                  relayEnvironment,
+                  savedAddress.createUserAddress.userAddressOrErrors.internalID,
+                  () => {
+                    onSuccess(savedAddress)
+                  },
+                  onError
+                )
+              } else {
+                onSuccess && onSuccess(savedAddress)
               }
-              const addressInput = convertShippingAddressToMutationInput(values)
-              if (createMutation) {
-                createUserAddress(
+
+              setCreateUpdateError(null)
+            }
+            const addressInput = convertShippingAddressToMutationInput(values)
+            if (modalAction.type === "createUserAddress") {
+              createUserAddress(
+                relayEnvironment,
+                addressInput,
+                handleSuccess,
+                handleError,
+                me,
+                closeModal
+              )
+            } else {
+              if (modalAction.type === "editUserAddress") {
```

Comment:
I would very much encourage you to flatten out all these conditionals into short circuiting returns (no nested, no else)

Reply:
https://preslav.me/2023/09/22/ditch-that-else/

Reply:
Good points - this was the last thing I did to get all the pieces working together again.